packages/core-data/src/actions.js

```diff
@@ -139,6 +144,58 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string}  kind              Kind of the deleted entity.
+ * @param {string}  name              Name of the deleted entity.
+ * @param {string}  recordId          Record ID of the deleted entity.
+ * @param {?Object} query             Special query parameters for the DELETE API call.
+ */
+export function* deleteEntityRecord( kind, name, recordId, query ) {
+	const entities = yield getKindEntities( kind );
+	const entity = find( entities, { kind, name } );
+	let error;
+	let deletedRecord = false;
+	if ( ! entity ) {
+		return;
+	}
+
+	yield {
+		type: 'DELETE_ENTITY_RECORD_START',
+		kind,
+		name,
+		recordId,
+	};
+
+	try {
+		let path = `${ entity.baseURL }/${ recordId }`;
+
+		if ( query ) {
+			path = addQueryArgs( path, query );
+		}
+
+		deletedRecord = yield apiFetch( {
+			path,
+			method: 'DELETE',
+		} );
+
+		yield removeItems( kind, name, recordId, true );
```

Comment:
This doesn't match the type signature of `removeItems`.

Reply:
updated the description of `removeItems` signature to get `{Array|number}` for `records`

Reply:
Doesn't that mean Array or number?

Reply:
that's what it means yes :)